Scenarios/EnablingTelemetry.md

```diff
@@ -1,58 +1,128 @@
+
+```
+"microsoftTelemetryConnectionString":  ""
+```
+
+By setting the Microsoft telemetry connection string to be an empty string you opt-out of sending basic telemetry to Microsoft. If on the other hand you want to send extended telemetry to Microsoft you can do that with the following setting.
 
 ```
 "SendExtendedTelemetryToMicrosoft" : true
 ```
 
-Each workflow starts with initialization task and ends with a postprocess task. During the initialization an operation Id(Guid) is generated and added to all the tasks in the workflow as ParentID. This property can be used to see all the signals sent for a workflow. The postprocess task sends the signal and duration of a workflow. Additionally, each task has its own signal and operationId. This could be used to investigate a task.
-
-Here is a list of the telemetry signals for different tasks:
-| Event ID | Description |
-| :-- | :-- |
-| DO0070 | AL-Go action ran: AddExistingApp |
-| DO0071 | AL-Go action ran: CheckForUpdates |
-| DO0072 | AL-Go action ran: CreateApp |
-| DO0073 | AL-Go action ran: CreateDevelopmentEnvironment |
-| DO0074 | AL-Go action ran: CreateReleaseNotes |
-| DO0075 | AL-Go action ran: Deploy |
-| DO0076 | AL-Go action ran: IncrementVersionNumber |
-| DO0077 | AL-Go action ran: PipelineCleanup |
-| DO0078 | AL-Go action ran: ReadSecrets |
-| DO0079 | AL-Go action ran: ReadSettings |
-| DO0080 | AL-Go action ran: RunPipeline |
-| DO0081 | AL-Go action ran: Deliver |
-| DO0082 | AL-Go action ran: AnalyzeTests |
-| DO0083 | AL-Go action ran: Sign |
-| DO0084 | AL-Go action ran: DetermineArtifactUrl |
-| DO0085 | AL-Go action ran: DetermineProjectsToBuild |
-
-Here is a list of the telemetry signals for different workflows:
-
-| Event ID | Description |
-| :-- | :-- |
-| DO0090 | AL-Go workflow ran: AddExistingAppOrTestApp |
-| DO0091 | AL-Go workflow ran: CICD |
-| DO0092 | AL-Go workflow ran: CreateApp |
-| DO0093 | AL-Go workflow ran: CreateOnlineDevelopmentEnvironment |
-| DO0094 | AL-Go workflow ran: CreateRelease |
-| DO0095 | AL-Go workflow ran: CreateTestApp |
-| DO0096 | AL-Go workflow ran: IncrementVersionNumber |
-| DO0097 | AL-Go workflow ran: PublishToEnvironment |
-| DO0098 | AL-Go workflow ran: UpdateGitHubGoSystemFiles |
-| DO0099 | AL-Go workflow ran: NextMajor |
-| DO0100 | AL-Go workflow ran: NextMinor |
-| DO0101 | AL-Go workflow ran: Current |
-| DO0102 | AL-Go workflow ran: CreatePerformanceTestApp |
-| DO0103 | AL-Go workflow ran: PublishToAppSource |
-| DO0104 | AL-Go workflow ran: PullRequestHandler |
+Sending extended telemetry to Microsoft is helpful for when we need to help investigate an issue in your repository.
+
+## Telemetry events and data
+
+AL-Go logs four different types of telemetry events: AL-Go action ran/failed and AL-Go workflow ran/failed. Each of those telemetry events provide slightly different telemetry but common dimensions for all of them are:
+
+**Common Dimensions**
```

Comment:
I thought the structure with the event ID and description was something we needed to follow, but I guess that is not the case?

Reply:
You mean these? https://github.com/microsoft/AL-Go/blob/8f5985dd807c532f1e2fdc03165259a2f7fb8f91/Actions/TelemetryHelper.psm1#L4

As for the description, we still log e.g. "AL-Go action ran: CreateApp" if it runs successfully. I did however change the message in case it fails to "AL-Go action failed: CreateApp". Hopefully that's okay :) If not, I can change it to always log "AL-Go action ran: "

The event IDs I got rid of though. IMO it's too easy to either forget to add them or misconfigure them. For example, right now we log telemetry for the Deploy Reference Documentation action as "AL-Go workflow ran: PublishToEnvironment" which means the telemetry gives a wrong picture. 

https://github.com/microsoft/AL-Go/blob/2b57fb3069d1fc2f724541638c48c21c49ab508e/Templates/AppSource%20App/.github/workflows/DeployReferenceDocumentation.yaml#L36

https://github.com/microsoft/AL-Go/blob/8f5985dd807c532f1e2fdc03165259a2f7fb8f91/Scenarios/EnablingTelemetry.md?plain=1#L48

Reply:
The Event ID pattern stems from this: https://learn.microsoft.com/en-us/dynamics365/business-central/dev-itpro/administration/telemetry-event-ids - not sure we need them - but everything in BC uses them

Reply:
One of the reasons could be for filtering on specific signals in telemetry. With the changes in this PR, you'd just have to filter on descriptions instead.